repos/config/detail/BUILD.bazel

```diff
@@ -11,15 +11,22 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 load("@python_deps//:requirements.bzl", "requirement")
 load("@ros2_config//:repos_lock_file.bzl", "REPOS_LOCK_FILE")
+load("@ros2_config//:repos_setup_file.bzl", "REPOS_SETUP_FILE")
+load("@ros2_config//:repos_overlay_files.bzl", "REPOS_OVERLAY_FILES")
 
 py_binary(
     name = "repos_lock.update",
     srcs = ["lock_repos.py"],
     main = "lock_repos.py",
-    data = ["@ros2//:ros2.repos", REPOS_LOCK_FILE],
-    args = ["$(execpath @ros2//:ros2.repos)", "$(execpath {})".format(REPOS_LOCK_FILE)],
+    data = ["generate_ros2_config.py", "@ros2//:ros2.repos", REPOS_LOCK_FILE, REPOS_SETUP_FILE] + REPOS_OVERLAY_FILES,
+    args = [
+        "$(execpath @ros2//:ros2.repos)",
+        "$(execpath {})".format(REPOS_LOCK_FILE),
+        "$(execpath {})".format(REPOS_SETUP_FILE),
```

Comment:
It is dirty hacked together. I will clean up if the approach, in general, is going the right way.

A user should provide the `setup.bzl` file from the client repo. During `repos_lock.update`, it will be populated with repository-loading content.

An example how it will look like on the client side:

```python
configure_ros2(
    distro = "humble",
    repo_index_overlays = ["@my_prj//thirdparty/rules_ros_vendor:overlay.repos"],  # optional overlay
    setup_file = "@my_prj//thirdparty/rules_ros_vendor:setup_ros_repositories.bzl",
)
```